addons/armory/functions/fnc_talkingBox.sqf

```diff
@@ -0,0 +1,30 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Mike
+ * Handles the possibility of talking Armory boxes, It's canon as of Halloween 2023.
+ *
+ * Arguments:
+ * 0: Armory Box <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_box] call FUNC(talkingBox);
+ *
+ * Public: No
+ */
+
+params ["_box"];
+
+// Handle random chance, 7%
+if (random 100 > 7) exitWith {};
+
+// Exit if ran out of lines
+if (GVAR(boxLines) isEqualTo []) exitWith {};
```

Comment:
If you implement config-based lines like in `radios` component, copy that array and delete items until it's empty. When it's empty, copy the original array again so you never run out of lines (auto-resetting).